### PR TITLE
Improve proxy shutdown

### PR DESCRIPTION
**Pull request by kubistika**
Reverted #5838, simply because it was a mistake in understanding how rdpNla*, rdpTls* instances are managed. After digging, I found that it is allocated in transport_connect_nla and freed by freerdp_disconnect. Furthermore, I found that the output from ASAN, that pointed to this leak, was actually caused because the proxy **itself** didn't close all active connections gracefully.

### CODE REVIEW

File: libfreerdp/core/nla.c [hunk 1]
```diff
@@ -2451,6 +2451,7 @@ void nla_free(rdpNla* nla)
 	sspi_SecBufferFree(&nla->ClientNonce);
 	sspi_SecBufferFree(&nla->PublicKey);
 	sspi_SecBufferFree(&nla->tsCredentials);
+	sspi_SecBufferFree(&nla->negoToken);
```
Comment (akallabeth) [inline libfreerdp/core/nla.c]: ouch, this will break redirect/gateway related stuff

File: libfreerdp/core/rdp.c [hunk 2]
```diff
@@ -1857,13 +1857,11 @@ void rdp_reset(rdpRdp* rdp)
 	license_free(rdp->license);
 	transport_free(rdp->transport);
 	fastpath_free(rdp->fastpath);
-	nla_free(rdp->nla);
```
Comment (akallabeth) [inline libfreerdp/core/rdp.c]: Same here, this is recreated to reset to initial state.

Comment (kubistika) [reply libfreerdp/core/rdp.c]: no, this is an addition i added in  #5838. is was a mistake, i explained why. this struct is created in transport_connect_nla and does not need to be recreated here

File: libfreerdp/core/nla.c (hunk 1 above)
Comment (kubistika) [reply libfreerdp/core/nla.c]: can you explain?
this buffer shouldn't be freed when the whole rdpNla* struct is freed?

Comment (kubistika) [reply libfreerdp/core/nla.c]: BTW I checked and xfreerdp ables to be redirected after this change, i didn't see anything wrong

File: libfreerdp/core/rdp.c (hunk 2 above)
Comment (akallabeth) [reply libfreerdp/core/rdp.c]: @kubistika oops, forgot ^^

File: libfreerdp/core/nla.c (hunk 1 above)
Comment (akallabeth) [reply libfreerdp/core/nla.c]: @kubistika no, this is most of the time just a pointer to another buffer and this will be a double free

Comment (kubistika) [reply libfreerdp/core/nla.c]: okay, reverted

**Pull request merged**